From a94c8b6f31d6618c13a75e84e76914cbb2312253 Mon Sep 17 00:00:00 2001
From: Elad Nachman <enachman@marvell.com>
Date: Wed, 18 Jun 2025 09:35:29 +0000
Subject: [PATCH 2/2] irqchip/irq-mvebu-gicp: clear pending irqs on init

For kexec case, left interrupt might generate spurious
interrupts in various A7/A8/CN913x interrupt system
from the I/O SB to the NB. Clear all pending interrupts
when the driver is initialized to prevent these spurious
interrupts.

Signed-off-by: Elad Nachman <enachman@marvell.com>
---
 patch/0019-mrvl-gicp-clr-irqs-on-probe.patch | 26 ++++++++++++++++++++
 patch/series                                 |  1 +
 2 files changed, 27 insertions(+)
 create mode 100644 patch/0019-mrvl-gicp-clr-irqs-on-probe.patch

diff --git a/patch/0019-mrvl-gicp-clr-irqs-on-probe.patch b/patch/0019-mrvl-gicp-clr-irqs-on-probe.patch
new file mode 100644
index 0000000..f9e7910
--- /dev/null
+++ b/patch/0019-mrvl-gicp-clr-irqs-on-probe.patch
@@ -0,0 +1,26 @@
+--- a/drivers/irqchip/irq-mvebu-gicp.c	2025-06-11 16:23:39.943033000 +0300
++++ b/drivers/irqchip/irq-mvebu-gicp.c	2025-06-11 16:35:15.706499500 +0300
+@@ -29,6 +29,7 @@
+ 
+ struct mvebu_gicp {
+ 	struct mvebu_gicp_spi_range *spi_ranges;
++	void __iomem *base;
+ 	unsigned int spi_ranges_cnt;
+ 	unsigned int spi_cnt;
+ 	unsigned long *spi_bitmap;
+@@ -244,6 +245,15 @@
+ 	}
+ 
+ 	platform_set_drvdata(pdev, gicp);
++	dev_err(&pdev->dev, "gicp - Clearing existing interrupts\n");
++	gicp->base = devm_platform_ioremap_resource(pdev, 0);
++	if (IS_ERR(gicp->base)) 
++		dev_err(&pdev->dev, "gicp - Cannot ioremap !\n");
++	else {
++		for (i = 0; i < 64; i++)
++			writel(i, gicp->base + GICP_CLRSPI_NSR_OFFSET);
++	} 
++	dev_err(&pdev->dev, "gicp - Cleared existing interrupts\n");
+ 
+ 	return 0;
+ }
diff --git a/patch/series b/patch/series
index c857b28..2b66008 100755
--- a/patch/series
+++ b/patch/series
@@ -184,6 +184,7 @@ cisco-npu-disable-other-bars.patch
 0016-arm64-dts-marvell-Add-Wistron-ES-1227-54TS-board.patch
 0017-Extend-driver-to-support-XMC-XM25QH256C-device.patch
 0018-i2c-mv64xxx-kdump-sonic.patch
+0019-mrvl-gicp-clr-irqs-on-probe.patch
 
 # amd-pensando elba support
 0000-Add-support-for-the-TI-TPS53659.patch
-- 
2.25.1

